File: src/SharedModels/Models/Equipment.cs
namespace SharedModels;

public enum EquipmentStatus
{
    AVAILABLE,
    UNDER_REPAIR,
    CHECKED_OUT,
    RETIRED,
}

public class Equipment
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public int Room { get; set; }
    public string PathToPhoto { get; set; } = null!;
    public EquipmentStatus Status { get; set; }
    public string Type { get; set; } = null!;
    public string SerialNumber { get; set; } = null!;
    public int SchoolId { get; set; }
    public School School { get; set; } = null!;
    public DateTime? UpdatedAt { get; set; }
    public DateTime? CreatedAt { get; set; }
    public ICollection<Request> Requests { get; set; } = new List<Request>();

}

File: src/sims_frontend/src/types/equipment.ts
import { EquipmentRequest } from "./request";

export enum EquipmentStatus {
    AVAILABLE = "AVAILABLE",
    UNDER_REPAIR = "UNDER_REPAIR",
    CHECKED_OUT = "CHECKED_OUT",
    RETIRED = "RETIRED",
}

export interface Equipment {
    id: number;
    name: string;
    room: number;
    pathToPhoto: string;
    status: EquipmentStatus;
    type: string;
    serialNumber: string;
    updatedAt?: string; // Using string for Date objects from JSON
    createdAt?: string; // Using string for Date objects from JSON
    requests: EquipmentRequest[]
}